cordformation/src/main/kotlin/net/corda/plugins/TypesafeUtils.kt

```diff
@@ -0,0 +1,38 @@
+package net.corda.plugins
+
+import java.util.regex.Pattern
+
+class TypesafeUtils {
+
+    companion object {
+
+        private val urlTokenPattern: Pattern = Pattern.compile("\\$\\{([^}]+)\\}")
+
+        /**
+         * Encodes a string that contains ${} placeholders so that can be parsed by Typesafe
+         */
+        fun encodeString(input: String): String {
+            val matcher = urlTokenPattern.matcher(input)
+            val builder = StringBuilder()
+            var endOfPrevToken = 0
+
+            while (matcher.find()) {
+                val startOfCurToken = matcher.start()
+                val token = matcher.group(1)
+                if (startOfCurToken != 0) {
+                    builder.append("\"${input.substring(endOfPrevToken, startOfCurToken)}\"")
```

Comment:
It's better to concatenate your strings using your existing `StringBuilder` here:
```kotlin
builder.append('\"').append(input.substring(endOfPrevToken, startOfCurToken)).append('\"')
```
Note that `StringBuilder` also allows you to add a single character.